task/powershell.ps1

```diff
@@ -8,32 +8,48 @@ Import-Module "$ScriptDir\Export-NUnitXml.ps1"
 $templatelocation = get-VstsInput -Name templatelocation -Require
 $resultlocation = get-VstsInput -Name resultLocation -Require
 
+function Test-FolderContents {
+    param(
+        [string]$folder,
+        [string]$filter
+    )
+    
+    #Path is always set to folder due to limitation of ARMTTK, filter then picks file(s) or full folder to test
+    $results=Test-AzTemplate -TemplatePath "$folder" -File "$filter" -ErrorAction Continue
+    Export-NUnitXml -TestResults $results -Path $resultlocation
+
+    if (!$results) { 
+        return 0
+    }
+        
+    return $($results.passed | Where-Object { $_ -eq $false } | Measure-Object).Count
+}
+
 ### Test Paths
-try{
-    Get-Item $templatelocation
+try {
+    $item = Get-Item $templatelocation
 }
-catch{
-    write-error "Template Location is not an existing folder, file or wildcard"
+catch {
+    Write-Error "Template Location is not an existing folder, file or wildcard"
 }
 #If a folder has been passed in set the template location to a wildcard for that folder
-if((Get-Item $templatelocation) -is [System.IO.DirectoryInfo]){
+if ($item -is [System.IO.DirectoryInfo]){
     $templatelocation = "$($templatelocation.Trimend('\'))\*"
 }
 
-$totalFileCount = $(Get-ChildItem $templatelocation).count
+$files = Get-ChildItem $templatelocation
+$totalFileCount = $files.count
 
-if($totalFileCount -lt 1){
+if ($totalFileCount -lt 1) {
     Write-Error "No files found in provided path"
 }
 
-### Run Tests
-$folder = split-path $templatelocation
-$filter = split-path $templatelocation -leaf
-#Path is always set to folder due to limitation of ARMTTK, filter then picks file(s) or full folder to test
-$results=Test-AzTemplate -TemplatePath "$folder" -File "$filter" -ErrorAction Continue
-Export-NUnitXml -TestResults $results  -Path $resultlocation
+$FailedNumber = 0
+foreach ($file in $files) {
+    $fileInfo = [System.IO.FileInfo]$file    
```

Comment:
If the folder that contains the arm templates itself has a folder in it, then this will throw an error. We should probably check that it is a file, and ignore if it is a folder. We could look at recursively going through folders, but lets keep it simple for now.

Reply:
Yes, that's a scenario I overlooked. Thanks for pointing it out.
I will try to address that as well.

Reply:
I think I solved it, by filtering the results of Get-ChildItem. 
We are only interested in json files, since the wildcard logic is entirely handled in the extension.
I also added the flag `-Recurse` to simplify specifying the template location(s).